flexbox/src/main/java/com/google/android/flexbox/FlexboxLayoutManager.java

```diff
@@ -1076,8 +1076,14 @@ private boolean updateAnchorFromPendingState(RecyclerView.State state, AnchorInf
         anchorInfo.mPosition = mPendingScrollPosition;
         anchorInfo.mFlexLinePosition = mFlexboxHelper.mIndexToFlexLine[anchorInfo.mPosition];
         if (mPendingSavedState != null && mPendingSavedState.hasValidAnchor(state.getItemCount())) {
-            anchorInfo.mCoordinate = mOrientationHelper.getStartAfterPadding() +
-                    savedState.mAnchorOffset;
+            if (isMainAxisDirectionHorizontal()) {
+                anchorInfo.mCoordinate = mOrientationHelper.getLayoutManager().getPaddingTop() +
+                        savedState.mAnchorOffset;
+            } else {
+                anchorInfo.mCoordinate = mOrientationHelper.getStartAfterPadding() +
+                        savedState.mAnchorOffset;
+            }
```

Comment:
I think this is not a correct way to fix the issue.

Actually some of the tests in FlexboxLayoutMangerTests are failing after patching this PR.

Couldn't look into the issue carefully, but I guess what needs to be cared is not whether the main axis direction is horizontal, but whether the  flex wrap is nowrap.